src/VGriveClient.vala

```diff
@@ -228,26 +228,41 @@ namespace App {
                 return false;
             }
         }
-        private void check_deleted_files () {
+        
+         private void check_deleted_files () {
             // Mira els fitxers que hi ha a la llibreria
             // Si no existeixen en local o en remot, el treu de la llibreria i l'elimina de on encara hi sigui
             if (!this.is_syncing ()) return;
+            
             var it = this.library.map_iterator ();
             bool exist_local, exist_remote, must_delete = false;
-            DriveFile remote_file;
+            DriveFile[] remote_files;
             string remote_id, filename, aux, lpath;
             Array<string> to_delete = new Array<string> ();
+            
             for (var has_next = it.next (); has_next; has_next = it.next ()) {
-                if (!this.is_syncing ()) return;
```

Comment:
Mantain this if. If the user has stopped the sync process we don't have to continue deleting files

Reply:
Right!